### ISSUE TITLE
Attaching a span to the trace started in the Tyk API Gateway from a grpc plugin when Open Telemetry is enabled.

### ISSUE THREAD

**Issue by ls-adrian-chong** [TykTechnologies/tyk#6489]
**Is your feature request related to a problem? Please describe.**
When Open Telemetry is enabled, I would like to be able to have a gRPC plugin add a span to the trace created by the Tyk API Gateway.   Also, add trace context details from the gRPC plugin.

Currently, there is no trace context provided to the gRPC plugin so if a new span is created from within the gRPC plugin, it is not a child of the parent trace started in the Tyk API Gateway, instead a whole new trace is created.

**Describe the solution you'd like**
A way to be able to attach a span to the parent trace created by the Tyk API Gateway from the gRPC plugin.
Suppose this could be the trace id sent as a field in the context.

Thinking setting details on the trace context would also be possible if the trace context was provided.

**Describe alternatives you've considered**
Enabling detailed tracing in the api definition - nothing from the gRPC plugin.
Using ddtrace from within the gRPC plugin - results in a new trace that is unconnected to the one started from the Tyk API Gateway.

Reference:
https://community.tyk.io/t/opentelemetry-trace-with-grpc-plugin/7695/2

**Comment by buger** [comment]
I think somthehing like this possible only with Go plugin, which has access to full native request context 🤔

**Comment by boser87** [comment]
As far as I understand, the tracing context [is not correctly passed](https://github.com/TykTechnologies/tyk/blob/2887a4aa6c19c2651643ed10d72e04429ec40401/gateway/coprocess_grpc.go#L50) within the Golang context as grpc metadata when using the grpc client?

**Comment by kercsoj** [comment]
Are there any chances to prioritize this feature request any time soon?

**Comment by buger** [comment]
/probe pls do you review, on complexity, or maybe some workaround already possible?

**Comment by buger** [comment]
/probe so basically in workaround solution, it is responsibility of the plugin to communicate to opentelemetry, and in the first solution it is more native, kind of? How this first solution, will look from the the view of gRPC plugin, can you give some example, lets say with Python? Will we need to modify protobuf too?

**Comment by buger** [comment]
/engineer can you try implement this idea? Do not forget to add the tests too

**Comment by kercsoj** [comment]
Based on Probe AI’s analysis, is there any chance this feature request can be prioritized in the Tyk Gateway backlog? Or, if an external contributor implements it in a fork, is there a realistic chance to getting it merged into the main Tyk Gateway repository?

**Comment by boser87** [comment]
It would be super nice to prio this one. Also, I don't think right now we have any workaround to reach the same behaviour with only using grpc middlewares or even native plugins right?

**Comment by forbidwolf** [comment]
I've opened a PR that implements this: #7813

The approach injects the active span's propagation headers into `Object.Metadata` in `BuildObject`, using the same global text map propagator (`otel.GetTextMapPropagator()`) that the gRPC transport layer already uses. This means plugins receive `traceparent` (W3C) or `x-b3-*` (B3) in `object.getMetadataMap()` — whichever format is configured via `context_propagation` — without any protobuf or interface changes.

Tested end-to-end against a real Java gRPC plugin: the `traceparent` header appears in `Object.Metadata` on every dispatched request and can be used directly to create child spans linked to the gateway trace.

Happy to adjust anything based on your feedback.

**Comment by forbidwolf** [comment]
@buger — following up on #7813, which implements this feature. The approach is minimal: inject the active span's propagation headers into `Object.Metadata` in `BuildObject` using the global text map propagator, no protobuf changes needed. We've tested it end-to-end with a real Java gRPC plugin and confirmed `traceparent` arrives in `object.getMetadataMap()` on every request.

Given the community interest here (and that others are blocked on this too), would you be able to take a look and merge if it looks good to you?
